File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res, maxWater = 0,0
        left , right = 0, len(height) - 1

        while left < right:
            width = right - left
            ht = min(height[left], height[right])

            res = width * ht
            maxWater = max(maxWater, res)

            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        
        return maxWater
        

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int ans = 0;
        while(left < right){
            int width = right - left;
            if(height[left] < height[right]){
                ans = Math.max(ans, height[left] * width);
                left++;
            }
            else{
                ans = Math.max(ans, height[right] * width);
                right--;
            }
        }
        return ans;
    }
}